Vagrantfile

```diff
@@ -319,23 +321,41 @@ Vagrant.configure("2") do |config|
         host.hostsupdater.aliases = info[:alias]
       end
 
-      host.vm.provider :virtualbox do |vb|
-        vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
-        vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
-        vb.customize ["modifyvm", :id, "--memory", info[:mem]]
-        vb.customize ["modifyvm", :id, "--name", hostname]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
-        vb.gui = info[:show_gui]
-        vb.name = hostname
+      if(vagrant_provider == 'virtualbox')
+        host.vm.provider :virtualbox do |vb|
+          vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
+          vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
+          vb.customize ["modifyvm", :id, "--memory", info[:mem]]
+          vb.customize ["modifyvm", :id, "--name", hostname]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
+          vb.gui = info[:show_gui]
+          vb.name = hostname
+        end
+      elsif(vagrant_provider == 'libvirt')
+        host.vm.provider :libvirt do |libvirt|
+          libvirt.cpus = info[:cpus]
+          libvirt.memory = info[:mem]
+          libvirt.nested = true
```

Comment:
Can you set VM names as we do with the virtualbox provider?

Reply:
yes i will set the default prefix to empty string